homeassistant/components/shelly/utils.py

```diff
@@ -8,24 +8,20 @@
 
 from homeassistant.components.sensor import DEVICE_CLASS_TIMESTAMP
 from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT
-from homeassistant.helpers import entity_registry
 
 from . import ShellyDeviceWrapper
+from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_remove_entity_by_domain(hass, domain, unique_id, config_entry_id):
-    """Remove entity by domain."""
-
+async def async_remove_shelly_entity(hass, domain, unique_id):
+    """Remove a Shelly entity."""
     entity_reg = await hass.helpers.entity_registry.async_get_registry()
-    for entry in entity_registry.async_entries_for_config_entry(
-        entity_reg, config_entry_id
-    ):
-        if entry.domain == domain and entry.unique_id == unique_id:
```

Comment:
unique_id check was a explicit request of @balloob on my previous commit.

Reply:
I have changed to implementaion which also checks `platform`:
```python
entity_id = entity_reg.async_get_entity_id(domain, DOMAIN, unique_id)
```
So this can only delete entities which are part of the Shelly integration, this is why it is not needed to compare `domain` and ` unique_id`